cedar-policy/src/ffi/utils.rs

```diff
@@ -345,6 +345,35 @@ impl Template {
             .add_template(template)
             .wrap_err(format!("failed to add template{msg} to policy set"))
     }
+
+    /// Get valid principals, actions, and resources.
+    pub fn get_valid_request_envs(
+        self,
+        s: Schema,
+    ) -> Result<
+        (
+            impl Iterator<Item = String>,
+            impl Iterator<Item = String>,
+            impl Iterator<Item = String>,
+        ),
+        miette::Report,
+    > {
+        let t = self.parse(None)?;
+        let (s, _) = s.parse()?;
+        let mut principals = Vec::new();
+        let mut actions = Vec::new();
+        let mut resources = Vec::new();
```

Comment:
should we use `HashSet` to avoid duplicates?

Reply:
Done. I used `BTreeSet` because the number of these types should not be large and we want to have an order.

Reply:
Makes sense to me